src/coreclr/vm/jitinterface.cpp

```diff
@@ -7868,6 +7868,38 @@ bool CEEInfo::haveSameMethodDefinition(
     return result;
 }
 
+CORINFO_CLASS_HANDLE CEEInfo::getTypeDefinition(CORINFO_CLASS_HANDLE type)
+{
+    CONTRACTL {
+        NOTHROW;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+    } CONTRACTL_END;
+
+    CORINFO_CLASS_HANDLE result = NULL;
+
+    JIT_TO_EE_TRANSITION();
+
+    TypeHandle constructedHandle(type);
+    TypeHandle definitionHandle = ClassLoader::LoadTypeDefThrowing(
+        constructedHandle.GetModule(),
+        constructedHandle.GetMethodTable()->GetCl(),
+        ClassLoader::ReturnNullIfNotFound,
```

Comment:
```suggestion
        ClassLoader::ThrowIfNotFound,
```

Reply:
Just curious -- are there cases where this can happen? (Presumably we'd be ok with OOM here, but we'd not want to start throwing during JIT because of an optimization if there are cases where the definition will fail to load even when the specific instantiation won't)

Reply:
It can happen for invalid IL or metadata. It is a general problem with optimizations and tiered compilation. Optimizations can change behavior for invalid IL or metadata.